libloki/api.js

```diff
@@ -225,61 +279,74 @@
       ourNumber,
       'private'
     );
-    const content = new textsecure.protobuf.Content({
-      pairingAuthorisation,
-    });
-    const isGrant = authorisation.primaryDevicePubKey === ourNumber;
-    if (isGrant) {
-      // Send profile name to secondary device
-      const lokiProfile = ourConversation.getLokiProfile();
-      // profile.avatar is the path to the local image
-      // replace with the avatar URL
-      const avatarPointer = ourConversation.get('avatarPointer');
-      lokiProfile.avatar = avatarPointer;
-      const profile = new textsecure.protobuf.DataMessage.LokiProfile(
-        lokiProfile
-      );
-      const profileKey = window.storage.get('profileKey');
-      const dataMessage = new textsecure.protobuf.DataMessage({
-        profile,
-        profileKey,
-      });
-      content.dataMessage = dataMessage;
-    }
     // Send
-    const options = { messageType: 'pairing-request' };
     const p = new Promise((resolve, reject) => {
-      const timestamp = Date.now();
-
-      const outgoingMessage = new textsecure.OutgoingMessage(
-        null, // server
-        timestamp,
-        [recipientPubKey], // numbers
-        content, // message
-        true, // silent
-        result => {
-          // callback
-          if (result.errors.length > 0) {
-            reject(result.errors[0]);
-          } else {
-            resolve();
-          }
-        },
-        options
+      const callback = result => {
+        // callback
+        if (result.errors.length > 0) {
+          reject(result.errors[0]);
+        } else {
+          resolve();
+        }
+      };
+      const pairingRequestMessage = textsecure.OutgoingMessage.buildPairingRequestMessage(
+        recipientPubKey,
+        ourNumber,
+        ourConversation,
+        authorisation,
+        pairingAuthorisation,
+        callback
       );
-      outgoingMessage.sendToNumber(recipientPubKey);
+
+      pairingRequestMessage.sendToNumber(recipientPubKey);
     });
     return p;
   }
 
+  function sendSessionRequestsToMembers(members = []) {
+    // For every member, see if we need to establish a session:
+    members.forEach(memberPubKey => {
```

Comment:
do we need to filter out ourselves?

Reply:
we could, but in all cases, the reloadDeviceAndSend methods now drops our current device: 
```js
reloadDevicesAndSend(number, recurse) {
    const ourNumber = textsecure.storage.user.getNumber();
    return () =>
      libloki.storage
        .getAllDevicePubKeysForPrimaryPubKey(number)
        // Don't send to ourselves
        .then(devicesPubKeys =>
          devicesPubKeys.filter(pubKey => pubKey !== ourNumber)
        )
        .then(devicesPubKeys => {
          if (devicesPubKeys.length === 0) {
            // No need to start the sending of message without a recipient
            return Promise.resolve();
          }
          return this.doSendMessage(number, devicesPubKeys, recurse);
        });
  },
```

Reply:
do you think we should remove ourself anyway